custom_libs/subliminal_patch/providers/jimaku.py

```diff
@@ -0,0 +1,316 @@
+        entry = data[0]
+        
+        entry_id = entry.get('id')
+        anilist_id = entry.get('anilist_id', None)
+        entry_name = entry.get('name')
+        
+        logger.info(f"Matched entry: ID: '{entry_id}', anilist_id: '{anilist_id}', name: '{entry_name}', english_name: '{entry.get('english_name')}'")
+        if entry.get("flags").get("unverified"):
+            logger.warning(f"This entry '{entry_id}' is unverified, subtitles might be incomplete or have quality issues!")    
+        
+        # Get a list of subtitles for entry
+        if isinstance(video, Episode):
+            episode_number = video.episode
+        
+        retry_count = 0
+        while retry_count <= 1:
+            retry_count += 1
+            
+            if isinstance(video, Episode):
+                url = f"entries/{entry_id}/files?episode={episode_number}"
+            else:
+                url = f"entries/{entry_id}/files"
+            data = self._get_jimaku_response(url)
+            
+            # Edge case: When dealing with a cour, episodes could be uploaded with their episode numbers having an offset applied
+            if not data and isinstance(video, Episode) and self.episode_number_override and retry_count < 1:
+                logger.warning(f"Found no subtitles for {episode_number}, but will retry with offset-adjusted episode number {video.series_anidb_episode_no}.")
+                episode_number = video.series_anidb_episode_no
+            elif not data:
+                return None
+        
+        # Filter subtitles
+        list_of_subtitles = []
+
+        archive_formats_blacklist = (".7z",) # Unhandled format
+        if not self.enable_archives:
+            disabled_archives = (".zip", ".rar")
+            
+            # Handle shows that only have archives uploaded
+            filter = [item for item in data if not item['name'].endswith(disabled_archives)]
+            if len(filter) == 0:
+                logger.warning("Archives are disabled, but only archived subtitles have been returned. Will therefore download anyway.")
+            else:
+                archive_formats_blacklist += disabled_archives
+
+        # Order subtitles to maximize quality
+        # Certain subtitle sources, such as Netflix/WebRip usually yield superior quality over others
+        try:
+            data = sorted(data, key=self._subtitle_sorting_key)
+        except Exception as e:
+            logger.warn(f"An error occurred while sorting subtitles: {e}")
+
+        for item in data:
+            subtitle_filename = item.get('name')
+            subtitle_url = item.get('url')
+
+            if not self.enable_ai_subs:
+                for ai_keyword in ["generated", "whisper"]:
+                    if re.search(r'\b' + re.escape(ai_keyword) + r'\b', subtitle_filename.lower()):
+                        logger.warning(f"Skipping AI generated subtitle '{subtitle_filename}'")
+                        continue
```

Comment:
This is actually very broad. There are episode/series titles that have this type word in their titles so this would give off a lot more false positives than expected. Just from this season we have "Whisper Me a Love Song", though right now [no files](https://jimaku.cc/entry/2874) use the substring. There are others such as [Whisper of the Heart](https://jimaku.cc/entry/744) which do.

We do a decent job of tagging these somewhat consistently though, I believe [`WhisperAI`](https://jimaku.cc/entry/391) (another example is [here](https://jimaku.cc/entry/372)) as a substring can get most of it. However this isn't perfect, [for example](https://jimaku.cc/entry/4417) this one isn't consistently tagged. It might be worth being consistent with it though.